### PR TITLE
[UT] mish op, conv+mish, fc+mish fuse passes

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
New features

### PR changes
OPs

### Describe
Add unit tests to the new UT framework for:
- mish operator,
- conv + mish fuse pass,
- fc + mish fuse pass.

### CODE REVIEW

**Review by jczaja — DISMISSED** [review]
LGTM

**Comment by jczaja** [comment]
@pawelpiotrowicz , @piotrekobiIntel  Please help with review

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import hypothesis
+from hypothesis import given, settings, seed, example, assume
+import hypothesis.strategies as st
+
+
+class TestConvMishMkldnnFusePass(PassAutoScanTest):
+    def is_program_valid(self, program_config: ProgramConfig) -> bool:
+        attrs = [
+            program_config.ops[i].attrs
+            for i in range(len(program_config.ops))
```
Comment (ghost) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py]:
```suggestion
            op.attrs
            for op in program_config.ops
```
Looks a bit cleaner in my opinion

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from program_config import TensorConfig, ProgramConfig
+import numpy as np
+import paddle.inference as paddle_infer
+from functools import partial
+from typing import Optional, List, Callable, Dict, Any, Set
+import unittest
+
+import hypothesis
+from hypothesis import given, settings, seed, example, assume
+import hypothesis.strategies as st
```
Comment (ghost) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py]: Some of these imports are unnecessary, as they are not used anywhere: paddle_infer, hypothesis and maybe some more. Please double check which ones are being used.
These unused imports are also present in the other test files.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
image
@jczaja @lidanqing-intel 
Does this mean that adding tests using the new framework requires approval? It seems like overkill to me and maybe it should be discussed with Baidu?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    def sample_predictor_configs(self, program_config):
+        config = self.create_inference_config(use_mkldnn=True)
+        yield config, ["conv2d"], (1e-5, 1e-5)
+
+    def test(self):
+        self.run_and_statis(quant=False, passes=["conv_mish_mkldnn_fuse_pass"])
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (jakpiase) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py:105]: Should we add paddle.enable_static() with this new UT framework?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import hypothesis.strategies as st
+
+
+class TestFCMishMkldnnFusePass(PassAutoScanTest):
+    def sample_program_config(self, draw):
+        x_shape = draw(
+            st.lists(
+                st.integers(
+                    min_value=1, max_value=8), min_size=2, max_size=5))
+        x_shape = [2, 1]
```
Comment (arlesniak) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py]: 28 line overrides the `x_shape value` that was set in line 24. Is it OK?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        x_rank = len(x_shape)
+        in_num_col_dims = draw(st.integers(min_value=1, max_value=x_rank - 1))
+        w_shape = draw(
+            st.lists(
+                st.integers(
+                    min_value=1, max_value=8), min_size=2, max_size=2))
+        w_shape[0] = int(np.prod(x_shape[in_num_col_dims:]))
+        w_shape = [1, 2]
```
Comment (arlesniak) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py]: ditto with line 31 and 35

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        fc_bias_shape = [w_shape[1], ]
+        if draw(st.booleans()):
+            fc_bias_shape.insert(0, 1)
+        fc_bias_shape = [2, ]
```
Comment (arlesniak) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py]: ditto for `fc_bias_shape`, is it OK ?

**Review by arlesniak — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py (hunk 4 above)
Comment (Silv3S) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py]: No, it's not okay for x_shape neither w_shape/bias_shape. It caused multiple runs on the same configuration. Thank you for your vigilance

**Comment by lidanqing-vv** [comment]
@baoachun Please review. This PR added required UTs. And this PR need approvals now

**Comment by jczaja** [comment]
@piotrekobiIntel It means that changing timeout (default one) for test execution require approval. I think @Silv3S modeled adding a test (in CMakeLists.txt) after other tests and each of them was setting this timeout.

**Comment by ghost** [comment]
> 

@Silv3S did you test if this timeout is required? Maybe it can be removed, that way this could be merged much quicker.

**Comment by Silv3S** [comment]
@piotrekobiIntel it works without extra properties but I wanted to follow the pattern from existing tests

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py (hunk 3 above)
Comment (baoachun) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv_mish_fuse_pass.py:105]: No, no need.

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            outputs=["mish_output"])
+        return program_config
+
+    def sample_predictor_configs(self, program_config):
+        config = self.create_inference_config(
+            use_mkldnn=True, passes=["fc_act_mkldnn_fuse_pass"])
+        yield config, ["fc"], (1e-5, 1e-5)
+
+    def test(self):
+        self.run_and_statis(quant=False, passes=["fc_act_mkldnn_fuse_pass"])
```
Comment (baoachun) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py:81]: Have we turned on the fc_act_mkldnn_fuse_pass pass in paddle_passs_builder.cc now?  If not, will this single test fail?

File: python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt [hunk 8]
```diff
@@ -133,5 +134,7 @@ if (WITH_MKLDNN)
   set_tests_properties(test_mkldnn_matmul_v2_transpose_reshape_fuse_pass PROPERTIES TIMEOUT 100)
   set_tests_properties(test_mkldnn_conv_transpose_bias_fuse_pass PROPERTIES TIMEOUT 100)
   set_tests_properties(test_conv_eltwiseadd_bn_fuse_pass PROPERTIES TIMEOUT 300)
+  set_tests_properties(test_mkldnn_conv_mish_fuse_pass PROPERTIES TIMEOUT 300)
```
Comment (baoachun) [inline python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt:137]: If the execution time of a UT does not exceed 300 seconds, there is no need to modify the single test timeout setting.

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py (hunk 7 above)
Comment (Silv3S) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_fc_mish_fuse_pass.py:81]: No, all fc passes are still disabled in _paddle_pass_builder.cc_. I tried to enable them but then unit test  _test_mkldnn_fc_act_fuse_pass_ failed. To make this working without modifying _paddle_pass_builder.cc_, pass name can be specified like: config -> passes =["fc_act_mkldnn_fuse_pass"] at line 77.

File: python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt (hunk 8 above)
Comment (Silv3S) [reply python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt:137]: Locally it works with or without explicitly setting timeout. Although not setting it caused CI fail on _test_mkldnn_conv_mish_fuse_pass_

**Comment by Silv3S** [comment]
@baoachun could you continue review? I think that these extra properties are necessary and we need approval

**Review by baoachun — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@XieYunshen, @chalsliu We need your approval to pass CI of this PR (fully reviewed by other reviewers). Please take a look. And Baoachun has reviewed.

**Review by chalsliu — APPROVED** [review]
LGTM for setting parameter TIMEOUT of UT.

**Comment by Silv3S** [comment]
@jczaja please merge

**Pull request merged**